python/pip_install/pip_repository.bzl

```diff
@@ -386,6 +386,10 @@ pip_repository_bzlmod_attrs = {
         default = False,
         doc = "Allow generating aliases in '@pip//:<pkg>' -> '@pip_<pkg>//:pkg'. This replaces the aliases generated by the `bzlmod` tooling.",
     ),
+    "repo_name": attr.string(
+        mandatory = True,
```

Comment:
If I am reading this correctly, putting this value as mandatory introduces a breaking change for our API.  Should we create a way not to break the API so that an upgrade path is more straightforward?  I am curious.

Reply:
According to this:

https://github.com/bazelbuild/rules_python/blob/c72c7bcf4e0899c275042328e8233e3124ccae86/python/pip_install/pip_repository.bzl#L414-L418

`pip_repository_bzlmod` is not part of API.

Reply:
👍 Thanks!  I am always learning.  I was wondering if it was private.  Thoughts on a follow-up PR for testing?

Reply:
I think that's a good idea. Let me figure it out later.